apps/emqx_management/src/emqx_mgmt_api_data_backup.erl

```diff
@@ -0,0 +1,361 @@
+                    )
+            }
+        }
+    };
+schema("/data/files/:filename") ->
+    #{
+        'operationId' => data_file_by_name,
+        get => #{
+            tags => ?TAGS,
+            desc => <<"Download a data backup file">>,
+            parameters => [
+                ?filename_field(true, #{in => path}),
+                ?node_field(false, #{in => query})
+            ],
+            responses => #{
+                200 => ?HOCON(binary),
+                400 => emqx_dashboard_swagger:error_codes(
+                    [?BAD_REQUEST], <<"Bad request">>
+                ),
+                404 => emqx_dashboard_swagger:error_codes(
+                    [?NOT_FOUND], <<"Backup file not found">>
+                )
+            }
+        },
+        delete => #{
+            tags => ?TAGS,
+            desc => <<"Delete a data backup file">>,
+            parameters => [
+                ?filename_field(true, #{in => path}),
+                ?node_field(false, #{in => query})
+            ],
+            responses => #{
+                204 => <<"No Content">>,
+                400 => emqx_dashboard_swagger:error_codes(
+                    [?BAD_REQUEST], <<"Bad request">>
+                ),
+                404 => emqx_dashboard_swagger:error_codes(
+                    [?NOT_FOUND], <<"Backup file not found">>
+                )
+            }
+        }
+    }.
+
+fields(files_response) ->
+    [
+        {data, ?ARRAY(?R_REF(backup_file_info))},
+        {meta, ?R_REF(emqx_dashboard_swagger, meta)}
+    ];
+fields(backup_file_info) ->
+    [
+        ?node_field(true),
+        ?filename_field(true),
+        {created_at,
+            ?HOCON(binary(), #{
+                desc => "Data backup file creation date and time",
+                required => true
+            })}
+    ];
+fields(import_request_body) ->
+    [?node_field(false), ?filename_field(true)];
+fields(data_backup_file) ->
+    [
+        ?filename_field(true),
+        {file,
+            ?HOCON(binary(), #{
+                desc => "Data backup file content",
+                required => true
+            })}
+    ].
+
+%%------------------------------------------------------------------------------
+%% HTTP API Callbacks
+%%------------------------------------------------------------------------------
+
+data_export(post, _Request) ->
+    case emqx_mgmt_data_backup:export() of
+        {ok, #{filename := FileName} = File} ->
+            {200, File#{filename => filename:basename(FileName)}};
+        Error ->
+            Error
+    end.
+
+data_import(post, #{body := #{<<"filename">> := FileName} = Body}) ->
+    case safe_parse_node(Body) of
+        {error, Msg} ->
+            {400, #{code => 'BAD_REQUEST', message => Msg}};
+        FileNode ->
+            CoreNode = core_node(FileNode),
+            response(
+                emqx_mgmt_data_backup_proto_v1:import_file(CoreNode, FileNode, FileName, infinity)
```

Comment:
Only core node is allowed to import a backup, because it implies using `mnesia:restore/2`. Calling it on a replicant node would mean that Mnesia data is only restored locally.